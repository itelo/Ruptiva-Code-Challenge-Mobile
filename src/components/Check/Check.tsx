import React from "react";
import { View } from "react-native";
import { ThemeType } from "../../utils/styles/ThemeContext";
import useStyles from "../../utils/styles/useStyles";

type CheckProps = {
 checked: boolean;
}

const Check = (props: CheckProps) => {
  const styles = useStyles(classes, props);
  return (
  <View style={styles.border}>
    <View style={styles.innerCircle} />
  </View>
  )
};

const classes = (theme: ThemeType, props: CheckProps) => ({
  border: {
    height: 20,
    width: 20,
    borderRadius: 10,
    borderColor: theme.palette.boring.blue,
    borderWidth: 1
  },
  innerCircle: {
    height: 14,
    width: 14,
    margin: 2,
    borderRadius: 10,
    backgroundColor: props.checked ? theme.palette.boring.blue : "#FFF"
  }
})

export default Check;
