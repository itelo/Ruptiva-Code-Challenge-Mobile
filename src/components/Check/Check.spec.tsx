import React from "react";
import Check from "./Check";
import { render } from "react-native-testing-library";

describe("<Check />", () => {
  it("test props checked={false}", () => {
    const { toJSON } = render(
      <Check checked={false} />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it("test props checked={true}", () => {
    const { toJSON } = render(
      <Check checked={true} />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});