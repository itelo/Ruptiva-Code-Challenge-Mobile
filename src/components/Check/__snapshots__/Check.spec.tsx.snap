// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Check /> test props checked={false} 1`] = `
<View
  style={
    Object {
      "borderColor": "#00b2ff",
      "borderRadius": 10,
      "borderWidth": 1,
      "height": 20,
      "width": 20,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 10,
          "height": 14,
          "margin": 2,
          "width": 14,
        },
        Object {
          "backgroundColor": "#FFF",
        },
      ]
    }
  />
</View>
`;

exports[`<Check /> test props checked={true} 1`] = `
<View
  style={
    Object {
      "borderColor": "#00b2ff",
      "borderRadius": 10,
      "borderWidth": 1,
      "height": 20,
      "width": 20,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 10,
          "height": 14,
          "margin": 2,
          "width": 14,
        },
        Object {
          "backgroundColor": "#00b2ff",
        },
      ]
    }
  />
</View>
`;
